src/project_types.rs

```diff
@@ -15,12 +22,32 @@ pub struct User {
     pub public_key: String,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, RustcEncodable, RustcDecodable)]
+pub struct SequenceSection {
+    pub seq_name: String,
+    pub index: u32,
+    pub num_frames: u32,
+    pub editor: Option<User>,
+    pub data: Vec<Vec<u8>>, // Row is channel, column is frame    
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, RustcEncodable, RustcDecodable)]
+pub struct Sequence {
+    pub name: String,
+    pub directory_name: String,
+    pub music_file_name: String,
+    pub music_duration_sec: u32,
+    pub frame_duration_ms: u32, // Default: 50ms
```

Comment:
Move these two comments to Sequence::new(). They make more sense there to clarify the Options, vs here (since they're not really part of the data type themselves).